modules/mesh_com/entrypoint.sh

```diff
@@ -1,62 +1,72 @@
-        gateway_ip="192.168.247.10" # FIXME: hardcoded for now. later detect automatically.
-        route add default gw $gateway_ip bat0
-        sleep 86400
-    elif [ "$DRONE_TYPE" == "groundstation" ]; then
-        DEFAULT_MESH_IP="192.168.248.1"
-        /opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
-        echo "mesh setup done"
-        sleep 86400
-    elif [ "$DRONE_TYPE" == "fog" ]; then
-        if [ "$MESH_CLASS" == "edge" ]; then
-            DEFAULT_MESH_IP="192.168.247.10"
-        else
-            # mesh class is gs
-            DEFAULT_MESH_IP="192.168.248.10"
-        fi
-
-        /opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
-        echo "mesh setup done"
-        if [ "$MESH_CLASS" == "gs" ]; then
-            gateway_ip="192.168.248.1" # FIXME: hardcoded for now. later detect automatically.
-            route add default gw $gateway_ip bat0
-        fi
-        sleep 86400
-    elif [ "$DRONE_TYPE" == "singlemesh" ]; then
-        # 192.168.248.11-192.168.248.253
-        DEFAULT_MESH_IP="192.168.248.$[ $RANDOM % 243 + 11 ]"
-        /opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
-        echo "mesh setup done"
-            # mesh class is gs
+if [ "$DRONE_TYPE" == "cm-recon" ]; then
+    DEFAULT_MESH_IP="192.168.$((RANDOM % 7 + 240)).$((RANDOM % 254 + 1))"
+    gateway_ip="192.168.247.10" # FIXME: hardcoded for now. later detect automatically.
+    
+elif [ "$DRONE_TYPE" == "groundstation" ]; then
+    DEFAULT_MESH_IP="192.168.248.1"
+    
+elif [ "$DRONE_TYPE" == "fog" ]; then
+    if [ "$MESH_CLASS" == "edge" ]; then
+        DEFAULT_MESH_IP="192.168.247.10"
+    elif [ "$MESH_CLASS" == "gs" ]; then
+        DEFAULT_MESH_IP="192.168.248.10"
         gateway_ip="192.168.248.1" # FIXME: hardcoded for now. later detect automatically.
-        route add default gw $gateway_ip bat0
-        sleep 86400
     else
-        echo "drone type not implemented: $DRONE_TYPE"
-        exit 1
-    fi
+			echo "Undefined mesh class"   
+	 fi 
+
+elif [ "$DRONE_TYPE" == "cm-fog" ]; then
+    DEFAULT_MESH_IP="192.168.248.$((RANDOM % 243 + 11))"
+    gateway_ip="192.168.248.1" # FIXME: hardcoded for now. later detect automatically.
+    
+else
+    echo "drone type not implemented: $DRONE_TYPE"
+    exit 1
+fi
+
+/opt/ros/${ROS_DISTRO}/share/bin/mesh-11s.sh $DEFAULT_MESH_MODE $DEFAULT_MESH_IP $DEFAULT_MESH_MASK $DEFAULT_MESH_MAC $DEFAULT_MESH_KEY $DEFAULT_MESH_ESSID $DEFAULT_MESH_FREQ $DEFAULT_MESH_TX $DEFAULT_MESH_COUNTRY
```

Comment:
mesh-executor should be enabled here. See row: https://github.com/tiiuae/mesh_com/pull/309/files#diff-31f2e5de9f3f4a563b757fc677f251e921209d421672608b52d862e6f73f6cf5R96